packages/api/src/ontology/Definition.ts

```diff
@@ -88,14 +88,32 @@ export interface PropertyDefinition {
   nullable?: boolean;
 }
 
-export interface ValidPropertyTypes {
+export interface ValidPropertyBaseTypes {
   string: string;
   datetime: Date;
   double: number;
   boolean: boolean;
   integer: number;
+  timestamp: Date;
+  short: number;
+  long: number;
+  float: number;
+  decimal: number;
+  byte: number;
+
+  attachment: any;
+  geopoint: any;
+  geoshape: any;
 }
 
+export type AttachmentTypes = {
```

Comment:
It felt like there are two ways to tackle this, one being have a `multiplicity` property on the property definition but then we'd need to infer the types on each one, or have string keys that map to the types. The reason I think multiplicity might not be the right solution here is we also will have a similar issue with TimeSeries which can have a string or a numeric subtype